torch/_tensor_str.py

```diff
@@ -65,132 +65,128 @@ def set_printoptions(
         PRINT_OPTS.linewidth = linewidth
 
 
-def _get_min_log_scale():
-    min_positive = float_info.min * float_info.epsilon  # get smallest denormal
-    if min_positive == 0:  # use smallest normal if DAZ/FTZ is set
-        min_positive = float_info.min
-    return math.ceil(math.log(min_positive, 10))
+class _Formatter(object):
+    def __init__(self, tensor):
+        tensor = tensor.view(tensor.nelement())
+        self.floating_dtype = tensor.dtype.is_floating_point
+        self.int_mode = True
+        self.sci_mode = False
+        self.max_width = 1
+
+        if self.floating_dtype:
+            for value in tensor.tolist():
+                value_str = '{}'.format(value)
+                self.max_width = max(self.max_width, len(value_str))
 
+        else:
+            try:
+                for value in tensor.tolist():
```

Comment:
this looks strange -- you are calling ceil when floating_dtype is false but not when it's true?  Am I missing something here?